components/organism/nestedCheckboxes/src/index.js

```diff
@@ -0,0 +1,118 @@
+import React, {useState} from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+
+import MoleculeCheckboxField from '@s-ui/react-molecule-checkbox-field'
+
+const baseClass = 'sui-OrganismNestedCheckboxes'
+
+const OrganismNestedCheckboxes = ({
+  id,
+  checkedIcon: CheckedIcon,
+  intermediateIcon: IntermediateIcon,
+  hideItemsIcon: HideItemsIcon,
+  items = [],
+  join = false,
+  labelParent = '',
+  onChangeItem = () => {},
+  onChangeParent = () => {},
+  showItems: showItemsProp = true,
+  showItemsIcon: ShowItemsIcon
+}) => {
+  const [showItems, setShowItems] = useState(showItemsProp)
+  const showToggleIcons = ShowItemsIcon !== null
+
+  const isParentFullChecked = items.every(({checked}) => checked === true)
+  const isParentHalfChecked =
+    !isParentFullChecked && items.some(({checked}) => checked === true)
+
+  const isParentFullOrHalfChecked = isParentFullChecked || isParentHalfChecked
+
+  return (
+    <ul className={baseClass}>
+      <li>
+        <MoleculeCheckboxField
+          id={id}
+          checked={isParentFullChecked}
+          checkedIcon={CheckedIcon}
+          intermediate={isParentHalfChecked}
+          intermediateIcon={IntermediateIcon}
+          label={labelParent}
+          onChange={() => onChangeParent(isParentFullOrHalfChecked)}
+          toggleIcon={showItems ? HideItemsIcon : ShowItemsIcon}
+          toggleIconOnChange={() => setShowItems(!showItems)}
+        />
+        {items.length > 0 && showToggleIcons && showItems && (
+          <ul className={`${baseClass}-list`}>
+            {items.map(item => {
+              const {id: childId, checked} = item
+
+              const listItemClass = `${baseClass}-listItem`
+              const listItemClassModifier = cx(listItemClass, {
+                [`${listItemClass}--joined`]: join
+              })
+
+              return (
+                <li key={childId} className={listItemClassModifier}>
+                  <MoleculeCheckboxField
+                    id={childId}
+                    checked={checked}
+                    checkedIcon={CheckedIcon}
+                    intermediateIcon={IntermediateIcon}
+                    onChange={e => onChangeItem(e.target)}
```

Comment:
I would recommend follow this for events callbacks: https://github.mpi-internal.com/scmspain/frontend-all-wiki/blob/c7974b262c69f92951fa83af3a21f299f638d582/wiki/DeveloperNotes/Recommendations-components-development.md#parameters-handled-by-events-callbacks

Reply:
Also for the `onChangeParent`.